crates/router/src/connector/zen/transformers.rs

```diff
@@ -402,7 +534,10 @@ fn get_item_object(
             Ok(ZenItemObject {
                 name: data.product_name.clone(),
                 quantity: data.quantity,
-                price: utils::to_currency_base_unit(data.amount, item.request.currency)?,
+                price: utils::to_currency_base_unit_with_zero_decimal_check(
```

Comment:
This is being used in cards and wallets PM, as discussed this is tested for cards and wallets

Reply:
@SamraatBansal While adding test cases for this PR, Let's include test cases of cards, wallets also